MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -1027,7 +1027,8 @@
 		// If this image is not yet available, extract and signal the first semaphore and fence.
 
 		signaler = _availabilitySignalers.front();
-		_availabilitySignalers.pop_front();
+    _availabilitySignalers.erase( _availabilitySignalers.begin() );
```

Comment:
I'm concerned about the performance implications of this. I think there's a reason this code originally used `std::list`. Vectors generally do not do well with insertion/removal from the front.

Reply:
@cdavis5e It really depends on the size of _availabilitySignalers. If this is smaller lets say than 20 elements, the new version should still be a bit faster as it avoids the heap allocation/deallocation which involves a system call. My initial thought was, that size of _availabilitySIgnalers will never exceed more than 4 elements as this list deals with swapchain images? If I am wrong, please revert back to the original version here.